frontend/src/test/java/com/google/tripmeout/frontend/servlet/PlaceParentServletTest.java

```diff
@@ -0,0 +1,222 @@
+                                                  .setTripId("123abc")
+                                                  .setPlaceId("ChIJ3S-JXmauEmsRUcIaWtf4MzE")
+                                                  .setLatitude(111.111)
+                                                  .setLongitude(45.4)
+                                                  .setUserMark(PlaceVisitModel.UserMark.MAYBE)
+                                                  .build();
+
+  @Before
+  public void setUp() {
+    initMocks(this);
+    Injector injector = Guice.createInjector(new GsonModelSerializationModule());
+    this.gson = injector.getInstance(Gson.class);
+  }
+
+  @Test
+  public void doPost_badUri_setsBadRequestStatus() throws IOException {
+    FakeHttpServletResponse response = new FakeHttpServletResponse();
+    PlaceVisitStorage placeStorage = new InMemoryPlaceVisitStorage();
+    PlaceParentServlet servlet = new PlaceParentServlet(placeStorage, gson);
+    when(request.getRequestURI()).thenReturn("/trips-number/abc123/placeVisits");
+    when(request.getReader())
+        .thenReturn(new BufferedReader(
+            new StringReader("{tripId: abc123, placeId: ChIJ3S-JXmauEmsRUcIaWtf4MzE}")));
+
+    servlet.doPost(request, response);
+
+    assertThat(response.getStatus()).isEqualTo(HttpServletResponse.SC_BAD_REQUEST);
+  }
+
+  @Test
+  public void doPost_badPlaceId_setsNotFoundStatus() throws IOException {
+    FakeHttpServletResponse response = new FakeHttpServletResponse();
+    PlaceVisitStorage placeStorage = new InMemoryPlaceVisitStorage();
+    PlaceParentServlet servlet = new PlaceParentServlet(placeStorage, gson);
+    when(request.getRequestURI()).thenReturn("/trips/abc123/placeVisits");
+    when(request.getReader())
+        .thenReturn(new BufferedReader(new StringReader("{tripId: abc123, placeId: hello}")));
+
+    servlet.doPost(request, response);
+    Optional<PlaceVisitModel> place = placeStorage.getPlaceVisit("abc123", "hello");
+
+    assertThat(place).isEmpty();
+    assertThat(response.getStatus()).isEqualTo(HttpServletResponse.SC_NOT_FOUND);
+  }
+
+  @Test
+  public void doPost_goodRequest_storesPlaceVisitModel()
+      throws IOException, ApiException, InterruptedException {
+    PlaceDetailsRequest placeRequest = new PlaceDetailsRequest(CONTEXT)
+                                           .placeId("ChIJ3S-JXmauEmsRUcIaWtf4MzE")
+                                           .fields(PlaceDetailsRequest.FieldMask.GEOMETRY_LOCATION,
+                                               PlaceDetailsRequest.FieldMask.NAME);
+
+    PlaceDetails details;
+    details = placeRequest.await();
+
+    PlaceVisitModel expected = PlaceVisitModel.builder()
+                                   .setTripId("abc123")
+                                   .setPlaceId("ChIJ3S-JXmauEmsRUcIaWtf4MzE")
+                                   .setUserMark(PlaceVisitModel.UserMark.YES)
+                                   .setName(details.name)
```

Comment:
Actually, it looks like we may not even need a mock service in this instance since the we're only calling it to get a place name and location. These should just hard-code fake/predetermined values.